endpoint/main.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/select.h>
+#include "connectivity/conn_http.h"
+#include "http_parser.h"
+#include "utils/crypto_utils.h"
+#include "utils/protocol.h"
+#include "utils/serializer.h"
+#include "utils/tryte_byte_conv.h"
+#include "utils/trytes.h"
+#include "utils/uart_utils.h"
+
+#define HOST "tangle-accel.puyuma.org"
+#define PORT "443"
+#define API "transaction/"
+#define SSL_SEED "nonce"
+#define ADDRESS                                                                \
+  "POWEREDBYTANGLEACCELERATOR999999999999999999999999999999999999999999999999" \
+  "999999A"
+#define ADDR_LEN 81
+#define MAX_MSG_LEN 1024
+
+#ifndef DEBUG
+#define MSG "%s:%s"
+#else
+#define MSG "%s:THISISMSG9THISISMSG9THISISMSG"
+#define ADDR_LOG_PATH "addr_log.log"
+#endif
+
+static char addr_log_template[] = "\n%s\n";
+
+int log_address(uint8_t *next_addr) {
+  FILE *fp;
+  char addr_log[ADDR_LEN + 3];
+  // Append the next address to the address log file
+  fp = fopen(ADDR_LOG_PATH, "a");
+  if (!fp) {
+    perror("open addr_log.log failed:");
+    fclose(fp);
+    return -1;
+  }
+  snprintf(addr_log, 83, addr_log_template, next_addr);
+  fputs(addr_log, fp);
+  fclose(fp);
+  return 0;
+}
+
+int main(int argc, char *argv[]) {
+  char tryte_msg[MAX_MSG_LEN] = {0}, msg[MAX_MSG_LEN] = {0}, addr[ADDR_LEN + 1] = ADDRESS,
+       ciphertext[MAX_MSG_LEN] = {0}, raw_msg[1000] = {0};
+  uint32_t raw_msg_len = 1 + ADDR_LEN + 20, ciphertext_len = 0, msg_len;
```

Comment:
What does `20` stand for?

Reply:
We can fix this with macro in the future PRs.